### ISSUE TITLE
hpctraceviewer has a bad error message when a database is not writeable

### ISSUE THREAD

**Issue by jmellorcrummey** [HPCToolkit/hpcviewer#32]
On Linux, hpctraceviewer fails to open a database and reports the error "Invalid thread access" when it tries to rewrite a fresh database of traces to create the slimmer callpath.xml file from experiment.xml. When I finally looked back at the shell where I launched hpctraceviewer, I found the report of an uncaught java error shown at the bottom of this report. I think the "Invalid thread access" comes from having the thread performing this operation crash and not be joinable. We should catch this exception and report "Error: database directory needs to be writable, but it is not"

```
java.io.FileNotFoundException: /home/tmp/hpctoolkit-database-2954/callpath.xml (Permission denied)
        at java.io.FileOutputStream.open0(Native Method)
        at java.io.FileOutputStream.open(FileOutputStream.java:270)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:101)
        at edu.rice.cs.hpc.data.util.Grep.removeMetrics(Grep.java:51)
        at edu.rice.cs.hpc.data.util.Grep.grep(Grep.java:33)
        at edu.rice.cs.hpc.data.experiment.xml.ExperimentFileXML.parse(ExperimentFileXML.java:170)
        at edu.rice.cs.hpc.data.experiment.LocalDatabaseRepresentation.open(LocalDatabaseRepresentation.java:41)
        at edu.rice.cs.hpc.data.experiment.BaseExperiment.open(BaseExperiment.java:161)
        at edu.rice.cs.hpc.data.experiment.ExperimentWithoutMetrics.open(ExperimentWithoutMetrics.java:23)
        at edu.rice.cs.hpc.traceviewer.data.controller.SpaceTimeDataController.<init>(SpaceTimeDataController.java:93)
        at edu.rice.cs.hpc.traceviewer.db.local.SpaceTimeDataControllerLocal.<init>(SpaceTimeDataControllerLocal.java:52)
        at edu.rice.cs.hpc.traceviewer.db.local.LocalDBOpener.openDBAndCreateSTDC(LocalDBOpener.java:93)
        at edu.rice.cs.hpc.traceviewer.db.TraceDatabase$JobOpeningDatabase.run(TraceDatabase.java:204)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)

```

**Comment by jmellorcrummey** [comment]
@laksono

**Comment by ptaffet** [comment]
I got this same error message when the experiment.xml file didn't exist.
The actual error message was:

    java.io.IOException: File does not exist or not readable: /Users/pat2/traces/impi2018.0-block/experiment.xml
	at edu.rice.cs.hpc.data.experiment.xml.ExperimentFileXML.parse(ExperimentFileXML.java:135)
	at edu.rice.cs.hpc.data.experiment.LocalDatabaseRepresentation.open(LocalDatabaseRepresentation.java:41)
	at edu.rice.cs.hpc.data.experiment.BaseExperiment.open(BaseExperiment.java:161)
	at edu.rice.cs.hpc.data.experiment.ExperimentWithoutMetrics.open(ExperimentWithoutMetrics.java:23)
	at edu.rice.cs.hpc.traceviewer.data.controller.SpaceTimeDataController.<init>(SpaceTimeDataController.java:93)
	at edu.rice.cs.hpc.traceviewer.db.local.SpaceTimeDataControllerLocal.<init>(SpaceTimeDataControllerLocal.java:52)
	at edu.rice.cs.hpc.traceviewer.db.local.LocalDBOpener.openDBAndCreateSTDC(LocalDBOpener.java:93)
	at edu.rice.cs.hpc.traceviewer.db.TraceDatabase$JobOpeningDatabase.run(TraceDatabase.java:204)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

**Comment by laksono** [comment]
Fixed in patch 7c14eb95b85b7f6b66f7543716653e10e8c75fe6
